config/m4/module.m4

```diff
@@ -0,0 +1,437 @@
+                                 [$2=skip],  dnl empty
+                                 [$2=dir     dnl DIR
+                                  _CHECK_MODULE_SET_DIR([$1], [$3], [$4], [$5],
+                                                        [$6], [$7])])
+          ])
+
+#
+# Configure module directories
+#
+# Usage: _CHECK_MODULE_SET_DIR([mod-dir], [inc], [incflag], [lib], [libflag], [if-failed])
+#
+m4_define([_CHECK_MODULE_SET_DIR],
+          [
+             $2=$1/include
+             $3=-I$1/include
+             m4_pushdef([dir_lib], [$4]) dnl use macro unwind
+             _CHECK_MODULE_FIXUP_LIB([$1], [$4], [$5=-L$dir_lib], [$6])
+             m4_popdef([dir_lib])
+          ])
+
+#
+# Run test if status is in allowed list
+#
+# Usage: _CHECK_MODULE_RUN([status], [allowed-statuses], [macro], [if-no-macro])
+#
+m4_define([_CHECK_MODULE_RUN],
+          [
+             mod_run_happy=no
+
+             m4_foreach([stat], [$2],
+                 [
+                     AS_IF([test "$1" = "stat"], [mod_run_happy=yes])dnl
+                 ])
+             AS_IF([test $mod_run_happy = yes], [$3], [$4])
+          ])
+
+#
+# Check if all directories are exist
+#
+# Usage: _CHECK_MODULE_IS_DIR([dir1, dir2, ...], [if-yes], [if-no])
+#
+AC_DEFUN([_CHECK_MODULE_IS_DIR],
+         [
+             AS_IF([test "x$1" != "x"], [module_check_dir_happy=yes], [module_check_dir_happy=no])dnl
+             m4_foreach([dir], [$1],
+                        [
+                            AS_IF([test -d "dir"], [], [module_check_dir_happy=no])dnl
+                        ])
+             AS_IF([test $module_check_dir_happy = yes], [$2], [$3])dnl
+         ])
+
+#
+# Fixup for lib path
+#
+# Usage: _CHECK_MODULE_FIXUP_LIB([lib], [dest_var], [if-success], [if-not])
+#
+AC_DEFUN([_CHECK_MODULE_FIXUP_LIB],
+         [
+             _CHECK_MODULE_IS_DIR(
+                 [$1/lib64],
+                 [$2=$1/lib64
+                  $3],
+                 [_CHECK_MODULE_IS_DIR([$1/lib],
+                                       [$2=$1/lib
+                                        $3], [$4])])
+         ])
+
+#
+# Check if argument is yes/no/guess/empty
+#
+# Usage: _CHECK_MODULE_IS_YNG([value], [if-yes], [if-no], [if-guess], [if-empty], [if-else])
+#
+AC_DEFUN([_CHECK_MODULE_IS_YNG],
+         [AS_CASE(["x$1"],
+                  ["xyes"],   [$2],
+                  ["xno"],    [$3],
+                  ["xguess"], [$4],
+                  ["x"],      [$5],
+                              [$6])
+         ])
+
+#
+# Check module header
+#
+# Usage: _CHECK_MODULE_HEADER([header], [cppflags], [if-found], [if-not-found], [includes])
+#
+AC_DEFUN([_CHECK_MODULE_HEADER],
+         [
+             mod_header_saved_CPPFLAGS=$CPPFLAGS
+             mod_header_happy=yes
+
+             AS_IF([test "x$2" != "x"], [CPPFLAGS="$2 $CPPFLAGS"])
+
+             m4_foreach([header], [$1],
+                 [ dnl have to clean check header cache to allov multiple checks
```

Comment:
Typo 'allow'. The same issue at lines 164, 196.

Reply:
yep, typo. fixed